### PR TITLE
Initial Commit

### PR DESCRIPTION
**Pull request by mayur-pitale**

### CODE REVIEW

**Comment by stayseesong** [comment]
@mayur-pitale is this still being drafted or is it ready for review?

**Comment by mayur-pitale** [comment]
@stayseesong - ready for review

File: src/connections/destinations/catalog/actions-s3/index.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
+---
+title: S3 (Actions) Destination
```
Comment (tcgilbert) [inline src/connections/destinations/catalog/actions-s3/index.md:2]:
```suggestion
title: AWS S3 (Actions) Destination
```

This is the integration name on our backend. If you need to update this lmk

**Review by tcgilbert — CHANGES_REQUESTED** [review]

**Review by tcgilbert — DISMISSED** [review]

File: src/connections/destinations/catalog/actions-s3/index.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+### Prerequisites
+Ensure you have the following in place before configuring the AWS S3 (Actions) destination:
+
+- Amazon S3 Bucket: Create a bucket in your AWS account or use an existing one where you want to store the event data.
+- AWS IAM Permissions: Verify that you have appropriate IAM roles with write access to the S3 bucket and permissions for the Segment connection.
+- IAM Access IDs: Prepare your AWS IAM ARN ID and IAM External ID. These will be needed to authenticate and authorize Segment with your S3 bucket.
+
+
+### Create an IAM role in the AWS console
+To setup the IAM role to properly authorize Segment with the AWS S3 (Actions) destination, follow the steps below.
```
Comment (stayseesong) [inline src/connections/destinations/catalog/actions-s3/index.md:41]:
```suggestion
To set up the IAM role to properly authorize Segment with the AWS S3 (Actions) destination:
```

File: src/connections/destinations/catalog/actions-s3/index.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+- IAM Access IDs: Prepare your AWS IAM ARN ID and IAM External ID. These will be needed to authenticate and authorize Segment with your S3 bucket.
+
+
+### Create an IAM role in the AWS console
+To setup the IAM role to properly authorize Segment with the AWS S3 (Actions) destination, follow the steps below. 
+
+1. Login to your AWS account 
+2. Create a new or use an existing bucket with `PutObject`, `GetObject`, `ListObject` access to the S3 bucket.
+3. Go to IAM > Roles > Create Role
+4. Provide the following policy permissions for the IAM that was just created:
```
Comment (stayseesong) [inline src/connections/destinations/catalog/actions-s3/index.md:46]:
```suggestion
1. Log in to your AWS account. 
2. Create a new or use an existing bucket with `PutObject`, `GetObject`, `ListObject` access to the S3 bucket.
3. Navigate to **IAM > Roles > Create Role**.
4. Provide the following policy permissions for the IAM that was just created: 
```

File: src/connections/destinations/catalog/actions-s3/index.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+            "Action": [
+                "s3:PutObject",
+                "s3:PutObjectAcl"
+            ],
+            "Resource": "arn:aws:s3:::<YOUR_BUCKET_NAME>/*"
+        }
+    ]
+}
+```
+5. Click on the Trust Relationships tab and edit the trust policy to allow the IAM user to assume the role (If a user is not already created, please refer to the AWS documentation to create a user)
```
Comment (stayseesong) [inline src/connections/destinations/catalog/actions-s3/index.md:63]:
```suggestion
5. Click on the Trust Relationships tab and edit the trust policy to allow the IAM user to assume the role. If a user is not already created, refer to the AWS documentation to create a user.
```

File: src/connections/destinations/catalog/actions-s3/index.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+       "Condition": {
+         "StringEquals": {
+           "sts:ExternalId": "<YOUR_EXTERNAL_ID>"
+         }
+       }
+     }
+   ]
+ }
+```
+## Add AWS S3 (Actions) Destination in Segment
```
Comment (stayseesong) [inline src/connections/destinations/catalog/actions-s3/index.md:85]:
```suggestion
## Step 2: Add the AWS S3 (Actions) Destination in Segment
```

File: src/connections/destinations/catalog/actions-s3/index.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+To finish configuration, enable the AWS S3 (Actions) Destination in your workspace.
```
Comment (stayseesong) [inline src/connections/destinations/catalog/actions-s3/index.md:86]:
```suggestion
To finish the configuration, enable the AWS S3 (Actions) Destination in your workspace. 
```

File: src/connections/destinations/catalog/actions-s3/index.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+    ![AWS S3](images/aws-s3-catalog.png)
+2. Select the data source you'll connect to the destination.
+3. Provide a unique name for the destination.
+4. Complete the destination settings:
+   1. Enter the name of the region in which the bucket you created above resides.
+   2. Enter the name of the bucket you created above. Be sure to enter the bucket's **name** and not URI.
+   3. Enter the ARN of the IAM role you created above. The ARN should follow the format `arn:aws:iam::ACCOUNT_ID:role/ROLE_NAME.`
+   4. Enter the IAM External ID, which is a value set in the Trust Relationship under your AWS IAM Role.
+5. Enable the destination.
+6. Verify Segment data is stored in the S3 bucket by navigating to the `<your_S3_bucket>/` in the AWS console.
```
Comment (stayseesong) [inline src/connections/destinations/catalog/actions-s3/index.md:98]:
```suggestion
1. Add the **AWS S3 (Actions)** destination from the Destinations tab of the catalog.
2. Select the data source you want to connect to the destination.
3. Provide a unique name for the destination.
4. Complete the destination settings:
   * Enter the name of the region in which the bucket you created above resides.
   * Enter the name of the bucket you created above. Be sure to enter the bucket's **name** and not URI.
   * Enter the ARN of the IAM role you created above. The ARN should follow the format `arn:aws:iam::ACCOUNT_ID:role/ROLE_NAME.`
   * Enter the IAM External ID, which is a value set in the Trust Relationship under your AWS IAM Role.
5. Enable the destination.
6. Verify the Segment data is stored in the S3 bucket by navigating to the `<your_S3_bucket>/` in the AWS console. 
```

**Review by stayseesong — CHANGES_REQUESTED** [review]

File: src/connections/destinations/catalog/actions-s3/index.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+   * Enter the name of the region in which the bucket you created above resides.
+   * Enter the name of the bucket you created above. Be sure to enter the bucket's **name** and not URI.
+   * Enter the ARN of the IAM role you created above. The ARN should follow the format `arn:aws:iam::ACCOUNT_ID:role/ROLE_NAME.`
+   * Enter the IAM External ID, which is a value set in the Trust Relationship under your AWS IAM Role.
+5. Enable the destination.
+6. Verify the Segment data is stored in the S3 bucket by navigating to the `<your_S3_bucket>/` in the AWS console. 
+
+### Actions
+Build your own Mappings. The following action is supported: 
+- [Sync to S3](#sync-to-s3)
```
Comment (stayseesong) [inline src/connections/destinations/catalog/actions-s3/index.md:98]: @mayur-pitale sorry, one last thing, where is this linking to? there is no #syn-to-s3 section on this page

**Comment by stayseesong** [comment]
@mayur-pitale also wanted to confirm, this integration is in beta?

**Comment by mayur-pitale** [comment]
> @mayur-pitale also wanted to confirm, this integration is in beta?

@stayseesong - I removed the Actions section. I believe including the actions-fields.html should include any actions connected to this destination, correct?

**Comment by stayseesong** [comment]
correct, adding actions-fields.html includes any actions connected to the destination. 

But is this integration in beta?

**Comment by mayur-pitale** [comment]
> correct, adding actions-fields.html includes any actions connected to the destination.
> 
> But is this integration in beta?

@stayseesong - Yes it is in private beta

**Review by stayseesong — APPROVED** [review]

**Pull request merged**